### ISSUE TITLE
Get this working with Php 7.x

### ISSUE THREAD

**Issue by BronsonQuick** [Chassis/xhprof#3]
I don't think Php 7.1+ is supported yet but we  should get this working with the other versions.

We might be able to use pecl to install this instead. Something like what JoHo has done on the HM AWS stack might work but using Puppet instead. e.g.
`curl -L https://github.com/humanmade/xhprof/archive/sampling-interval.zip > /tmp/xhprof.zip && unzip -o /tmp/xhprof.zip -d /tmp && cd /tmp/xhprof-sampling-interval/extension && phpize && ./configure && make && make install && echo "extension=xhprof.so" > /etc/php/7.0/fpm/conf.d/20-xhprof.ini && echo "extension=xhprof.so" > /etc/php/7.0/cli/conf.d/20-xhprof.ini`

**Comment by joehoyle** [comment]
7.1+ should be supported, I think i've installed it under 7.1. However < 7 won't work, that would need a different extension / fork. THere's no version on pecl that's sufficient, you need to use the humanmaded one linked above!

Is Chassis dropping support for 5.6 at some point?

**Comment by BronsonQuick** [comment]
Awesome! Thanks for letting me know about 5.6 and 7.1! I'll add some logic to handle a different fork for 5.6.

>Is Chassis dropping support for 5.6 at some point?

We support 5.6 -> 7.2 at the moment. We used to have support for 5.2+ but ondrej dropped support the ppa for php-oldstable which had 5.2 - 5.5. Currently we default to 7.0.x (x being the latest minor release).

I'll just drop a couple of notes here for myself for Monday too. I've just manually installed this to see if I could get it running. When I write the puppet package I'll need to make sure it has php-pear and php7.0-dev as dependencies. Using the above script it'll install `xhprof.so` into `/tmp/xhprof-sampling-interval/extension/modules` so we'll want to copy that into `/usr/lib/php/20151012` which is where we're installing our modules to on Chassis then we'll need to notify `php7.0-fpm` so that it restarts.
